infrastructure/code/code.go

```diff
@@ -215,18 +215,27 @@ func (sc *Scanner) Scan(ctx context.Context, path string, folderPath string) (is
 func internalScan(ctx context.Context, sc *Scanner, folderPath string, logger zerolog.Logger, filesToBeScanned map[string]bool) (results []snyk.Issue, err error) {
 	span := sc.BundleUploader.instrumentor.StartSpan(ctx, "code.ScanWorkspace")
 	defer sc.BundleUploader.instrumentor.Finish(span)
+	ctx, cancel := context.WithCancel(span.Context())
+	defer cancel()
+
+	t := progress.NewTracker(true)
+	// monitor external tracker & context cancellations
+	go func() { t.CancelOrDone(cancel, ctx.Done()) }()
+
+	t.BeginWithMessage("Snyk Code: scanning "+folderPath, "starting scan")
+	defer t.EndWithMessage("Snyk Code: scan of " + folderPath + " done")
 
 	fileFilter, _ := sc.fileFilters.Load(folderPath)
 	if fileFilter == nil {
 		fileFilter = filefilter.NewFileFilter(folderPath, &logger)
 		sc.fileFilters.Store(folderPath, fileFilter)
 	}
-	files := fileFilter.FindNonIgnoredFiles()
+	files := fileFilter.FindNonIgnoredFiles(t)
 
-	if sc.useIgnoresFlow() {
-		results, err = sc.UploadAndAnalyzeWithIgnores(span.Context(), folderPath, files, filesToBeScanned)
+	if sc.useIgnoresFlow() && !t.IsCanceled() && ctx.Err() == nil {
```

Comment:
move this check outside of if statement so that it doesnt go to else branch if canceled